app/upgrades/v2/upgrades_test.go

```diff
@@ -0,0 +1,105 @@
+package v2_test
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/suite"
+	"github.com/tharsis/evmos/v2/app"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	abci "github.com/tendermint/tendermint/abci/types"
+	tmproto "github.com/tendermint/tendermint/proto/tendermint/types"
+	feemarkettypes "github.com/tharsis/ethermint/x/feemarket/types"
+)
+
+type UpgradeTestSuite struct {
+	suite.Suite
+
+	ctx sdk.Context
+	app *app.Evmos
+}
+
+func (suite *UpgradeTestSuite) SetupTest() {
+	// setup feemarketGenesis params
+	feemarketGenesis := feemarkettypes.DefaultGenesisState()
+	feemarketGenesis.Params.EnableHeight = 1
+	feemarketGenesis.Params.NoBaseFee = false
+
+	suite.app = app.Setup(false, feemarketGenesis)
+	suite.ctx = suite.app.BaseApp.NewContext(false, tmproto.Header{
+		ChainID: "evmos_9001-2",
+	})
+}
+
+func TestUpgradeTestSuite(t *testing.T) {
+	suite.Run(t, new(UpgradeTestSuite))
+}
+
+const upgradeHeight = 58700
+
+func (suite *UpgradeTestSuite) TestUpdateClaims() {
+	testCases := []struct {
+		msg        string
+		preUpdate  func()
+		update     func()
+		postUpdate func()
+		expPass    bool
+	}{
+		{
+			"Test Claims module migrations",
+			func() {
+				claimsParams := suite.app.ClaimsKeeper.GetParams(suite.ctx)
+				claimsParams.EnableClaims = false
+				suite.app.ClaimsKeeper.SetParams(suite.ctx, claimsParams)
+				suite.Require().False(suite.app.ClaimsKeeper.GetParams(suite.ctx).EnableClaims)
+			},
+			func() {
+				suite.ctx = suite.ctx.WithBlockHeight(upgradeHeight)
+				suite.Require().NotPanics(func() {
+					beginBlockRequest := abci.RequestBeginBlock{}
+					suite.app.BeginBlocker(suite.ctx, beginBlockRequest)
```

Comment:
is this technically calling the migration?

Reply:
...kind of? 

BeginBlocker calls BeginBlockForks

Reply:
yes this is moving through the migration logic with a height gated fork, because it runs upgrade, and it then runs the upgrade handler, which calls `RunMigrations`

given that this is `v2_test`, this is the test that makes sense to run.